tests/ui/coroutine/return-types-diverge.rs

```diff
@@ -0,0 +1,18 @@
+// compile-flags: --edition 2024 -Zunstable-options
+// check-pass
+
+#![feature(gen_blocks)]
+
+async gen fn async_gen_fn() -> i32 { return }
+
+gen fn gen_fn() -> i32 { return }
+
+fn async_gen_block() {
+    async gen { yield (); return };
+}
+
+fn gen_block() {
+    gen { yield (); return };
```

Comment:
To test divergence, shouldn't these panic or something rather than return unit as bare `return` does?

Reply:
`return` evaluates to the type `!`:

https://github.com/rust-lang/rust/blob/3f28fe133475ec5faf3413b556bf3cfb0d51336c/compiler/rustc_hir_typeck/src/expr.rs#L778

Reply:
Got it. Thanks for the explanation! 🙂